scripts/builder/checker_builder/config.py

```diff
@@ -0,0 +1,104 @@
+"""
+Configuration file logic for Checker builds
+"""
+import logging
+import math
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+import yaml
+from jinja2 import Environment, PackageLoader, select_autoescape
+from jinja2.environment import Template
+from marshmallow import Schema, ValidationError, fields
+from marshmallow.decorators import post_load
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+DEFAULT_CONFIG = Path("checker.yaml")
+
+# ================================================================================================
+# Config classes
+# ================================================================================================
+
+# TODO: Additional support for setting certain parameters at build time can be implemented
+# using:
+# def commented(comment: str, default=None) -> Field:
+#     """Creates a dataclass field with a comment in its metadata"""
+#     return field(default=default, metadata={"comment": comment})
+# @dataclass
+# class SystemParameters:
+#     """Checker system parameters"""
+
+#     fminting: Decimal = commented(
+#         "Dimensionless. Factor used for setting the minting limit.",
+#         default=(Decimal(21) / Decimal(10)),
+#     )
```

Comment:
Right, this one has to go as well. I am still thinking about the parameters actually, since their types can vary and I am not sure what's the best way to get them from a file (even though it would be nice for people deploying checker not having to look at the OCaml code at all). We would definitely need to avoid floating point numbers from python, and other things that could create inconsistencies. I guess passing strings would be the safest (though totally untyped, of course). :thinking: This is all for another PR of course.

Reply:
I think that we should* actually be pretty safe in this regard still. If someone specifies a decimal like `"2.1"` we can extract that as a fixedpoint (`Decimal`) in Python and convert it to a fraction so that it ultimately is rendered as a `ratio` in the generated source (i.e. the same way that they are currently implemented in `constants.ml`)